vowpalwabbit/core/include/vw/core/reduction_stack.h

```diff
@@ -21,9 +19,9 @@ class default_reduction_stack_setup : public setup_base_i
 
   void delayed_state_attach(VW::workspace& all, VW::config::options_i& options) override;
 
-  // this function consumes all the reduction_stack until it's able to construct a base_learner
+  // this function consumes all the reduction_stack until it's able to construct a learner
   // same signature as the old setup_base(...) from parse_args.cc
-  VW::LEARNER::base_learner* setup_base_learner() override;
+  VW::LEARNER::learner* setup_base_learner() override;
```

Comment:
It seems like this should return a shared_pointer as well?

Reply:
Added comments clarifying this along with a wrapper function that returns shared_ptr

Reply:
Actually, with the other changes made this now returns shared_ptr